service/posts/views.py

```diff
@@ -11,20 +11,12 @@ class PublicPostsList(generics.ListAPIView):
     pagination_class = PostsPagination
     serializer_class = PostSerializer
     authentication_classes = (NodeBasicAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    # No permission class
 
     def get_queryset(self):
-        node = self.request.user
-
-        if not node.share_posts:
-            return Post.objects.none()
-
-        queryset = Post.objects.filter(visibility="PUBLIC")
+        remote_node = self.request.user
```

Comment:
It's set in NodeBasicAuthentication. The first return value of `authenticate_credentials()` there is what populates `request.user`. More info here: http://www.django-rest-framework.org/api-guide/authentication/#custom-authentication (`authenticate()` eventually calls `authenticate_credentials()` and returns its return value, which you can see here: https://github.com/encode/django-rest-framework/blob/master/rest_framework/authentication.py#L86)